tidy3d/components/simulation.py

```diff
@@ -1058,6 +1061,42 @@ def _validate_monitor_size(self) -> None:
                 f"a maximum of {MAX_SIMULATION_DATA_SIZE_GB:.2f}GB are allowed."
             )
 
+    def _validate_modes_size(self) -> None:
+        """Warn if mode sources or monitors have a large number of points."""
+
+        def warn_mode_size(monitor: AbstractModeMonitor, msg_header: str, custom_loc: List):
+            """Warn if a mode component has a large number of points."""
+            num_cells = np.prod(self.discretize_monitor(monitor).num_cells)
+            if num_cells > WARN_MODE_NUM_CELLS:
+                consolidated_logger.warning(
+                    msg_header + f"has a large number ({num_cells:1.2e}) of grid points. "
+                    "This can lead to solver slow-down and increased cost. "
+                    "Consider making the size of the component smaller, as long as the modes "
+                    "of interest decay by the plane boundaries."
```

Comment:
I think you forgot `custom_loc=custom_loc` here

Reply:
Ah, good catch! Is there a good way to test that the `custom_loc` is set correctly? Can it be added to the automatic tests?